sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1817,3 +1817,24 @@ def test_sparse_manhattan_readonly_dataset(csr_container):
     Parallel(n_jobs=2, max_nbytes=0)(
         delayed(manhattan_distances)(m1, m2) for m1, m2 in zip(matrices1, matrices2)
     )
+
+
+# Some deprecated scipy metrics raise DeprecationWarning warning, filter here
+@pytest.mark.filterwarnings("ignore:The .* metric is deprecated since SciPy .*")
+@pytest.mark.parametrize("metric", PAIRWISE_BOOLEAN_FUNCTIONS)
+@pytest.mark.parametrize(
+    "pairwise_fn", [pairwise_distances_argmin, pairwise_distances_argmin_min]
+)
+def test_pairwise_argmin_correct_warnings_for_bool_and_nonbool(metric, pairwise_fn):
+    X = [[True, True, False, True], [True, False, True, True]]
+    Y = [[False, True, False, True], [True, False, True, True]]
+    X_arr = np.asarray(X)
+    Y_arr = np.asarray(Y)
+    with warnings.catch_warnings(record=True) as record:
+        pairwise_fn(X, Y, metric=metric)
+        assert len(record) == 0, f"A DataConversionWarning was raised for {metric}"
```

Comment:
Could we maybe give more info for someone seeing this message in a pytest run, e.g., no warning should have been raised for boolean inputs for {metric} ?

Also, technically we do not check that the warning raised was a `DataConversionWarning`, just that a warning was raised?